### ISSUE TITLE
BUG: Failure at Execution.cpp:242/getStaticAddr()! in GenMC 0.7, IMM and RC11

### ISSUE THREAD

**Issue by db7** [MPI-SWS/genmc#37]
I am getting some weird error in a rather simple code with mallocs.

```
BUG: Failure at Execution.cpp:242/getStaticAddr()!
```

And sometimes
```
The allocating operation (malloc()) does not happen-before the memory access!
```

[Here is a gist](https://gist.github.com/db7/1aa157778ffa814adeabeb8cdfcc769b) that can reproduce this cases. 

Interestingly the behavior or imm changes with -disable-race-detection. I thought race detection was completely disabled with IMM by default.

**Comment by db7** [comment]
I think this example I sent is representative. We had similar issues elsewhere. It is not only about the unexpected changes in behavior when using `-disable-race-detection` with IMM. That was actually a side note.

The issue is rather the bug itself. I think I didnt write in the ticket but just in the gist:
```c
#ifdef ENQUEUE_NONATOMIC
	br->br_ring[prod_head] = buf;
#else
	__atomic_store_n(&br->br_ring[prod_head], buf, __ATOMIC_RELAXED);
#endif
```

If the access is non-atomic, it crashes with `BUG` warning or `malloc does not happen-before` depending on whether I am using IMM or RC11. From my understanding, both cases should work with non-atomic accesses as well. I don't think these are related to LKMM (missing consistency checks #35), but it feels like the code shouldn't fail with non-atomics.

**Comment by michaliskok** [comment]
Should be fixed in v0.8. (Also, detection of memory races is no longer disabled by `-disable-race-detection`.)
